modules/@apostrophecms/doc-type/index.js

```diff
@@ -554,17 +554,9 @@ module.exports = {
       // taking into account issues like relationship fields keeping their data in
       // a separate ids property, etc.
       fieldsPresent(input) {
-        const schema = self.schema;
-        const output = [];
-        for (const field of schema) {
-          if (field.type.name.substring(0, 5) === '_relationship') {
-            if (_.has(input, field.idsStorage)) {
-              output.push(field.name);
-            }
-          } else {
-            output.push(field.name);
-          }
-        }
+        return self.schema
+          .filter((field) => input[field.name])
```

Comment:
This is just checking for truthiness which is not enough. `0` is a legitimate value. `_.has` should be used.

Reply:
Right! Done